sys/entropy_source/adc_noise/Kconfig

```
@@ -0,0 +1,74 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+menuconfig KCONFIG_MODULE_ENTROPY_SOURCE_ADC_NOISE
+    bool "Configure ADC Noise entropy source module"
+    depends on MODULE_ENTROPY_SOURCE_ADC_NOISE
+    help
+        Configure the ADC Noise entropy source using Kconfig.
+
+if KCONFIG_MODULE_ENTROPY_SOURCE_ADC_NOISE
+
+choice
+    bool "ADC default sampling resolution"
+    default ENTROPY_SOURCE_ADC_RES_10BIT
+    help
+        This parameter sets the ADC sampling resolution. Please note that not all
+        platforms support every value.
+
+config ENTROPY_SOURCE_ADC_RES_6BIT
+    bool "6 Bit"
+
+config ENTROPY_SOURCE_ADC_RES_8BIT
+    bool "8 Bit"
+
+config ENTROPY_SOURCE_ADC_RES_10BIT
+    bool "10 Bit"
+
+config ENTROPY_SOURCE_ADC_RES_12BIT
+    bool "12 Bit"
+
+config ENTROPY_SOURCE_ADC_RES_14BIT
+    bool "14 Bit"
+
+config ENTROPY_SOURCE_ADC_RES_16BIT
+    bool "16 Bit"
+
+endchoice
+
+config ENTROPY_SOURCE_ADC_LINE_NUM
+    int "ADC line"
+    range 0 16
+    default 0
+    help
+        The ADC line maps to an I/O pin. This number acts as index to an array
+        of predefined ADC devices that contain the pin definition. Typically,
+        the array is defined by a board in a periph_conf.h file. Please note that
+        a board is not required to specify a minimum number of lines.
+
+config ENTROPY_SOURCE_ADC_HMIN
+    int "Estimated entropy per sample [2^16 * bit/sample]"
+    range 1 524288
+    default 65536
```

```suggestion
    default 0
```

Having a non-zero default value is a foot gun. Setting the default to zero will force a application developers or board maintainers to override this with a (hopefully) well justified choice.

I bet 99% of the users never read the help message and the warning it contains and just go with the default value. Making the default value defunct will force them to spent the time required on this.

---

This needs to be changed in the header file as well, maybe there even a warning could be added when the default value is used?